lib/FetchEnvs.js

```diff
@@ -14,235 +14,195 @@
+const KIND_MAP = new Map([
+  ['secretKeyRef', 'Secret'],
+  ['secretMapRef', 'Secret'],
+  ['configMapRef', 'ConfigMap'],
+  ['configMapKeyRef', 'ConfigMap']
+]);
+
 module.exports = class FetchEnvs {
+
+  get [Symbol.toStringTag]() {
+    return 'FetchEnv';
+  }
+
   constructor(controllerObject) {
     if (!controllerObject) {
       throw Error('FetchEnvs must have: controller object instance');
     }
     this.data = controllerObject.data;
-    this.namespace = objectPath.get(this.data, 'object.metadata.namespace');
+    this.namespace = this.data?.object?.metadata?.namespace;
     this.kubeResourceMeta = controllerObject.kubeResourceMeta;
     this.kubeClass = controllerObject.kubeClass;
-
+    this.api = this.kubeResourceMeta.request.bind(this.kubeResourceMeta);
     this.updateRazeeLogs = controllerObject.updateRazeeLogs ?
       ((logLevel, log) => { controllerObject.updateRazeeLogs(logLevel, log); }) :
       (() => { log.debug('\'updateRazeeLogs()\' not passed to fetchEnvs. will not update razeeLogs on failure to fetch envs'); });
   }
 
-  async _lookupDataReference(valueFrom) {
-    let result;
-    let type;
-    let uri = '';
-    let key = '';
-    let kubeError;
-    if (valueFrom['secretKeyRef']) {
-      let secretName = objectPath.get(valueFrom, 'secretKeyRef.name');
-      let namespace = objectPath.get(valueFrom, 'secretKeyRef.namespace', this.namespace);
-      key = objectPath.get(valueFrom, 'secretKeyRef.key');
-      uri = `/api/v1/namespaces/${namespace}/secrets/${secretName}`;
-      let res;
+  #secretMapRef(conf) {
+    return this.#genericMapRef(conf, 'secretMapRef', true);
+  }
+
+  #secretKeyRef(conf) {
+    return this.#genericKeyRef(conf, 'secretKeyRef', true);
+  }
+
+  #configMapRef(conf) {
+    return this.#genericMapRef(conf, 'configMapRef');
+  }
+
+  #configMapKeyRef(conf) {
+    return this.#genericKeyRef(conf, 'configMapKeyRef');
+  }
+
+  async #genericMapRef(conf, valueFrom = 'genericMapRef', decode = false) {
+    let resource;
+    let kubeError = ERR_NODATA;
+    const ref = conf[valueFrom];
+    const optional = !!conf.optional;
+
+    const {
+      apiVersion = 'v1',
+      kind = KIND_MAP.get(valueFrom),
+      namespace = this.namespace,
+      name
+    } = ref;
+
+    const krm = await this.kubeClass.getKubeResourceMeta(apiVersion, kind, 'update');
+
+    if (krm) {
       try {
-        res = await this.kubeResourceMeta.request({ uri: uri, json: true });
-      } catch (e) {
-        log.warn(e);
-        kubeError = e.message;
+        resource = await krm.get(name, namespace);
+      } catch (error) {
+        kubeError = error.message;
       }
-      if (res && objectPath.has(res, ['data', key])) {
-        result = Buffer.from(objectPath.get(res, ['data', key]), 'base64').toString();
+    }
+
+    const data = resource?.data;
+
+    if (!data) {
+      const msg = `failed to get envFrom ${JSON.stringify(conf)}. ${kubeError}`;
```

Comment:
consistent with other log
```suggestion
      const msg = `failed to get envFrom: ${JSON.stringify(conf)}. ${kubeError}`;
```